exploration/hist_monitor_plotting.md

```diff
@@ -0,0 +1,357 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.1
+  kernelspec:
+    display_name: ds-aa-cub-hurricanes
+    language: python
+    name: ds-aa-cub-hurricanes
+---
+
+# Historical monitor plotting
+<!-- markdownlint-disable MD013 -->
+Plotting historical triggers based on historical forecasts
+
+```python
+%load_ext jupyter_black
+%load_ext autoreload
+%autoreload 2
+```
+
+```python
+import ocha_stratus as stratus
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+
+from src.constants import *
+```
+
+## Load and merge data
+
+```python
+# temporarily load IBTrACS locally since DB is down
+filename = "temp/ibtracs.NA.list.v04r01.csv"
+df_ibtracs = pd.read_csv(filename, skiprows=[1])
+```
+
+```python
+df_storms = (
+    df_ibtracs.groupby("SID")[["NAME", "USA_ATCF_ID"]].last().reset_index()
+)
+df_storms = df_storms.rename(
+    columns={"SID": "sid", "NAME": "name", "USA_ATCF_ID": "atcf_id"}
+)
+df_storms = df_storms[df_storms["sid"].str[:4].astype(int) >= 2000]
+df_storms["atcf_id"] = df_storms["atcf_id"].str.lower()
+```
+
+```python
+df_storms
+```
+
+```python
+blob_name = f"{PROJECT_PREFIX}/processed/impact/emdat_cerf_upto2024.parquet"
+df_impact = stratus.load_parquet_from_blob(blob_name)
+df_impact["cerf"] = ~df_impact["Amount in US$"].isnull()
+cols = [
+    "sid",
+    "cerf",
+    "Total Affected",
+    "Total Deaths",
+    "Total Damage, Adjusted ('000 US$)",
+    "Amount in US$",
+]
+df_impact = df_impact[cols]
+```
+
+```python
+df_impact.loc[df_impact["sid"] == IKE, "Total Affected"] = 2.6e6
+```
+
+```python
+blob_name = blob_name = (
+    f"{PROJECT_PREFIX}/processed/nhc/monitors_nhc_chirpsgefs.parquet"
+)
+df_monitors = stratus.load_parquet_from_blob(blob_name)
+```
+
+```python
+df_monitors_max = (
+    df_monitors.groupby(["atcf_id", "lt_name"]).max().reset_index().dropna()
+)
+```
+
+```python
+df_monitors_max
+```
+
+```python
+df_monitors_max["year"] = df_monitors_max["atcf_id"].str[-4:].astype(int)
+```
+
+```python
+n_years = (
+    df_monitors_max["issue_time"].max().year
+    - df_monitors_max["issue_time"].min().year
+    + 1
+)
+```
+
+```python
+target_rps = {
+    "action": 3.7,
+    "readiness": 3,
+}
+```
+
+```python
+target_trig_years = {
+    lt_name: int((n_years + 1) / rp) for lt_name, rp in target_rps.items()
+}
+```
+
+```python
+target_trig_years
+```
+
+```python
```

Comment:
perhaps we could add just a bit more markdown text to this document. Explaining that we are just looking at `action` in the plots (could even be added to plot titles), could also explain why we are just looking at q50 in plots, I think it's because it had the best correlation w/ `Total Affected` but wouldn't hurt to add a bit of text on that point as well